### ISSUE TITLE
geocode always returns blank - special character issue?

### ISSUE THREAD

**Issue by apukwa** [airhadoken/twitter-lib#13]
When using Geocode no results ever come back. 

Example twitter search string:  "#PoMo AND geocode:49.287907,-122.852812,100km"

Which gets translated as:  

https://api.twitter.com/1.1/search/tweets.json?lang=en&count=5&since_id=0&include_entities=false&result_type=recent&q=%2523PoMo%20AND%20geocode%3A49.287907%252C-122.852812%252C100km

(even without the hashtag it doesn't work).  Wondering if the commas are properly formatted (maybe they should be %2C instead of %252C?).

**Comment by airhadoken** [comment]
Can you try calling `.fetch()` with this url and the commas changed to `%2C`?

OAuth1's and TwitterLib's  `.fetch()` work like `UrlFetchApp.fetch()` so you'll need to call `.getContentText()` on the returned response object to see what you got back from Twitter.

**Comment by apukwa** [comment]
That did it... single-escaped commas (%2C) work properly whereas double-escaped commas fail.  Thanks so much for all your hard work on this!

**Comment by apukwa** [comment]
Fat fingers, sorry.

**Comment by airhadoken** [comment]
Great.  I've added this as a v23 milestone, so it will go into the next release.

As a workaround for now, to get the same effect as `.searchTweets()` (minus the tweet processor function), you can call `JSON.parse(<your twitterlib instance>.fetch(<your url>).getContentText("UTF-8")).statuses`

---

EDIT: previous version of this comment incorrectly said that the top level key of the returned object was "data" -- should be "statuses"

**Comment by apukwa** [comment]
Thanks kindly!  I'll do that now.

**Comment by apukwa** [comment]
Alas, doesn't seem to work.  Split it up just to make sure the fetch works -- the logger shows results on line 3, showing the fetch was okay, but "tweetsX" remains undefined after line 4.

```
  var twit = new Twitter.OAuth(props);
  var response = twit.fetch("https://api.twitter.com/1.1/search/tweets.json?lang=en&count=5&since_id=0&include_entities=false&result_type=recent&q=%2523PoMo%20AND%20geocode%3A49.287907%2C-122.852812%2C100km");
  Logger.log(response.getContentText("UTF-8"));
  var tweetsX = JSON.parse(response.getContentText("UTF-8")).data;
  Logger.log(tweetsX[0]);
```

**Comment by apukwa** [comment]
I suspect instead of .data I need .statuses.  Going to give it a go.

**Comment by airhadoken** [comment]
I edited my previous comment above; it should be `statuses` and not data.

**Comment by apukwa** [comment]
All's well for now, I'll revert to .searchTweets on the next run so I can use the processing.  :)

Thanks again for this, as well as for your hard work in getting it going!

**Comment by airhadoken** [comment]
You're welcome.  I'm glad to see it's stable & useful enough to have a user base!

**Comment by airhadoken** [comment]
Closed by #20
